src/site/_collections/algolia.js

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {createHash} = require('crypto');
+const striptags = require('striptags');
+const stripcomments = require('strip-comments');
+const removeMarkdown = require('remove-markdown');
+
+const AuthorsCollection = require('./authors');
+const TagsCollection = require('./tags');
+const {defaultLocale} = require('../_data/site');
+const {livePosts} = require('../_filters/live-posts');
+const {getDefaultUrl} = require('../_filters/urls');
+const {generateSrc} = require('../_includes/components/Img');
+const {supportedLocales} = require('../../../shared/locale');
+
+/**
+ * Cleans text and shrink the size of the given fulltext to fit within a certain limit,
+ * at the nearest found newline character.
+ *
+ * @param {string} content
+ * @param {number} [limit]
+ * @return {string}
+ */
+function limitText(content, limit = 7500) {
+  // Remove all comments
+  content = stripcomments(content);
+  // Remove all html
+  content = striptags(content);
+  // Remove all markdown
+  content = removeMarkdown(content);
+  // Remove extra spaces
+  content = content.replace(/\s+/g, ' ');
+
+  if (content.length <= limit) {
+    return content;
+  }
+
+  // Find the nearest prior newline to the 10k limit.
+  let newlineIndex = content.lastIndexOf('\n', limit);
+  if (newlineIndex === -1) {
+    newlineIndex = limit;
+  }
+  return content.slice(0, newlineIndex);
+}
+
+/**
+ * @param {EleventyCollectionObject} collections
+ * @returns {AlgoliaCollection}
+ */
+module.exports = (collections) => {
+  /** @type {AlgoliaCollection} */
+  const algoliaCollection = [];
+  /** @type {{
+    [url: string]: {
+      [locale: string]: AlgoliaCollectionItem;
+    };
+  }} */
+  const algoliaCollectionProcessing = {};
+  // if (process.env.ELEVENTY_ENV !== 'prod') {
+  //   return algoliaCollection;
+  // }
+
+  /**
+   * @param {AlgoliaCollectionItem} algoliaCollectionItem
+   */
+  const addToCollection = (algoliaCollectionItem) => {
```

Comment:
Can you write a comment explaining what all this function is doing? I'm having a hard time understanding what's going on.